src/Sulu/Bundle/RouteBundle/DependencyInjection/PageTreeCompilerPass.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\RouteBundle\DependencyInjection;
+
+use Sulu\Bundle\RouteBundle\Content\Type\PageTreeRouteContentType;
+use Sulu\Component\Content\Metadata\BlockMetadata;
+use Sulu\Component\Content\Metadata\PropertyMetadata;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+class PageTreeCompilerPass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container)
+    {
+        $structureFactory = $container->get('sulu_page.structure.factory');
+
+        $pageTreeEnabled = false;
+
+        foreach ($structureFactory->getStructureTypes() as $structureType) {
+            foreach ($structureFactory->getStructures($structureType) as $metadata) {
+                if ($this->hasPageTree($metadata->getProperties())) {
+                    $pageTreeEnabled = true;
+
+                    break 2;
+                }
+            }
+        }
+
+        $container->setParameter('sulu_route.page_tree_enabled', $pageTreeEnabled);
```

Comment:
If I read that code correctly we are only setting this to true if one of the pages is using a `page_tree_route`, right? Does that save any performance? Or how does that justify this code? Asking because it is kind of an anti-pattern to rely on services in a compiler pass, we are having troubles in other places because of that.

Reply:
I think this is only used to enable/disable the MovePageTreeCommand